libcudacxx/test/utils/nvidia/nvrtc/nvrtcc_run.h

```diff
@@ -65,7 +65,7 @@ static void load_and_run_gpu_code(const std::string inputFile, const RunConfig&
 
   CUDA_SAFE_CALL(cuInit(0));
   CUDA_SAFE_CALL(cuDeviceGet(&cuDevice, 0));
-  CUDA_SAFE_CALL(cuCtxCreate(&context, 0, cuDevice));
+  CUDA_SAFE_CALL(cuDevicePrimaryCtxRetain(&context, cuDevice));
```

Comment:
Both the old code and the new code are confusing -- I think after retrieving the primary context we also need to set it to current.
```python
>>> driver.cuInit(0)
(<CUresult.CUDA_SUCCESS: 0>,)
>>> err, ctx = driver.cuDevicePrimaryCtxRetain(0)
>>> err
<CUresult.CUDA_SUCCESS: 0>
>>> ctx
<CUcontext 0x563a77dae570>
>>> driver.cuCtxGetCurrent()  # no context is current
(<CUresult.CUDA_SUCCESS: 0>, <CUcontext 0x0>)         
>>> driver.cuCtxPushCurrent(ctx)
(<CUresult.CUDA_SUCCESS: 0>,)
>>> driver.cuCtxGetCurrent()  # the primary context is now current
(<CUresult.CUDA_SUCCESS: 0>, <CUcontext 0x563a77dae570>)
```
Unless, by the time this function is called, the caller has already initialized the context (perhaps through cudart)?